test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -328,6 +536,22 @@ rm -rf "$top" && mkdir "$top"
 cd "$top"
 
 
+if [ "$two_versions" = true ]; then
+    # Build the branches
```

Comment:
Running test/checkpoint in other options is properly controlled with the versions it is supported. In case if the user-provided version doesn't support the test/checkpoint, we are throwing errors very late. How about throwing the error here before building the branches if any of the branches is not supported test checkpoint?

Reply:
That's a good suggestion. Will add the version check logic here. 
What I'm not sure is if the version check logic in the `run_test_checkpoint()` and `verify_test_checkpoint()` could be removed. Will need to run a few tests to confirm (with the `="-p" command option).